### PR TITLE
[gpt3] add constants, compliance rules and initial RCP data

### PR DESCRIPTION
**Pull request by sgpyc**

### CODE REVIEW

**Review by anmolgupt — CHANGES_REQUESTED** [review]

**Comment by ShriyaPalsamudram** [comment]
@sgpyc - can you please fix all review comments so it can be merged this week?

**Comment by sgpyc** [comment]
@anmolgupt & @ShriyaPalsamudram Thanks for the review. Changed the PR accordingly in commit [8060d14](https://github.com/mlcommons/logging/pull/303/commits/8060d140e41afb1e2246abc3e8571e1a8a63dfb0).

File: mlperf_logging/rcp_checker/training_3.0.0/rcps_gpt3.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+
+  "gpt3_ref_1536":
+  {
+    "Benchmark": "gpt3",
+    "Creator": "Google & NVIDIA",
+    "When": "Prior to 3.0 submission",
+    "Platform": "TPU-v4-1536 / PaxML, 64 A100-80GB / Megatron-LM",
```
Comment (ShriyaPalsamudram) [inline mlperf_logging/rcp_checker/training_3.0.0/rcps_gpt3.json:8]: 1024 A100-80GB GPUs were used for all RCPs

Comment (sgpyc) [reply mlperf_logging/rcp_checker/training_3.0.0/rcps_gpt3.json]: Thanks for the info. Updated the PR.

**Review by ShriyaPalsamudram — DISMISSED** [review]

**Review by nv-rborkar — DISMISSED** [review]

**Comment by nv-rborkar** [comment]
@erichan1 Could you please merge this too? I don't seem to have write access.

**Comment by itayhubara** [comment]
Found one issue: please add 'gpt3' to the array in line 5 of the closed_common.yaml https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/compliance_checker/training_3.0.0/closed_common.yaml#L5

**Comment by itayhubara** [comment]
another issue: in closed_bert.yaml we required to log train**ed**_samples and in common.yaml we require train_samples. I think we should just remove the 'trained_samples" key from closed bert.yaml https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/compliance_checker/training_3.0.0/closed_gpt3.yaml#L65

**Comment by itayhubara** [comment]
one more bug missing apostrophes in line 20 of closed_gpt3.yaml https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/compliance_checker/training_3.0.0/closed_gpt3.yaml#L20

 CHECK: " v['value'] == 2e-5 or v['value**'**] == 3e-5 "

**Comment by itayhubara** [comment]
Additionally, we should:

1.  change xlsx_config.yaml in the results summarize (https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/result_summarizer/xlsx_config.yaml#L35) :

    "3.0.0":
      bert: ["Benchmark results (minutes)", "NLP", "Wikipedia", "BERT"]
      gpt3: ["Benchmark results (minutes)", "LLM", "C4", "GPT3"]
      dlrm_v2: ["Benchmark results (minutes)", "Recommendation", "1TB Clickthrough", "DLRM"]
      maskrcnn: ["Benchmark results (minutes)", "Object detection, heavy-weight", "COCO", "Mask R-CNN"]
      minigo: ["Benchmark results (minutes)", "Reinforcement Learning", "Go", "Minigo"]
      resnet: ["Benchmark results (minutes)", "Image classification", "ImageNet", "ResNet"]
      ssd: ["Benchmark results (minutes)", "Object detection, light-weight", "OpenImages", "RetinaNet"]
      rnnt: ["Benchmark results (minutes)", "Speech recognition", "LibriSpeech", "RNN-T"]
      unet3d: ["Benchmark results (minutes)", "Image segmentation (medical)", "KiTS19", "3D U-Net"]
      default: [" ", " ", " "]

2. Add gpt3 key to the submission_runs dictionary at the rcp_checker.py https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/rcp_checker/rcp_checker.py#L21
3. Update benchmark_meta.py as suggested in Shriya's PR:  https://github.com/mlcommons/logging/pull/306

**Comment by erichan1** [comment]
@itayhubara @nv-rborkar tag @pgmpablo157321 when you need to merge. I also don't have write access.

**Comment by sgpyc** [comment]
> another issue: in closed_gpt3.yaml we required to log train**ed**_samples and in common.yaml we require train_samples. I think we should just remove the 'trained_samples" key from closed_gpt3.yaml https://github.com/sgpyc/logging/blob/gpt3/mlperf_logging/compliance_checker/training_3.0.0/closed_gpt3.yaml#L65

`train_samples` is ambiguous: it can mean the training dataset size or number of samples used when reaching target accuracy. `trained_samples` would only mean the latter.

**Comment by sgpyc** [comment]
@itayhubara thanks for the detail reviews. Pushed ec8ca0b with changes accordingly.

**Review by ShriyaPalsamudram — APPROVED** [review]

**Review by nv-rborkar — APPROVED** [review]

**Pull request merged**